app/src/main/java/com/mybus/model/FavoriteLocation.java

```diff
@@ -0,0 +1,88 @@
+package com.mybus.model;
+
+import io.realm.RealmObject;
+import io.realm.annotations.PrimaryKey;
+import io.realm.annotations.Required;
+
+/**
+ * Created by Julian Gonzalez <jgonzalez@devspark.com>
+ * <p/>
+ * RealmObject to persist favorite locations
+ */
+public class FavoriteLocation extends RealmObject {
+    @PrimaryKey
+    private String name;
```

Comment:
It looks like with realm the only way is declaring the name as primary key (the unique constraint is not supported yet). In that case the name should be a read-only property for the object (the setter should be removed).